Objective-C/CBLArray.m

```diff
@@ -42,89 +42,104 @@ - (instancetype) initWithArray: (NSArray*)array {
 }
 
 
-- /*internal */ (instancetype) initWithFleeceData: (CBLFLArray*)data {
-    self = [super initWithFleeceData: data];
-    if (self ) {
-        _array = [NSMutableArray array];
-        [self loadBackingFleeceData];
-    }
-    return self;
-}
-
-
 #pragma mark - GETTER
 
 
 - (nullable id) objectAtIndex: (NSUInteger)index {
+    if (!_array) {
+        id value = [super objectAtIndex: index];
+        if ([value isKindOfClass: [CBLReadOnlyDictionary class]] ||
+            [value isKindOfClass: [CBLReadOnlyArray class]]) {
+            [self copyFleeceData];
```

Comment:
We need to copyFleeceData because we need to cache the converted Dictionary and Array so that it can return the same instance if users calling to get the object second time.